### PR TITLE
Theme Showcase: Add Translatabilty

### PR DESCRIPTION
**Pull request by akirk**
#### Changes proposed in this Pull Request

* Fix a crash when an empty theme is returned from the API.
* Make the Theme filter names transatable.

Before:

Screenshot 2019-11-15 at 21 14 22

Screenshot 2019-11-15 at 21 14 27

After:

Screenshot 2019-11-15 at 21 01 58

Screenshot 2019-11-15 at 21 12 10

### CODE REVIEW

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx [hunk 1]
```diff
@@ -85,6 +85,24 @@ class MagicSearchWelcome extends React.Component {
 			}
 		);
 
+		const taxonomyTranslations = {
+			feature: i18n.translate( 'Feature', {
```
Comment (a8ci18n) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **36** times:
`translate( 'Feature' )` *ES Score: 9*


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx [hunk 2]
```diff
@@ -85,6 +85,24 @@ class MagicSearchWelcome extends React.Component {
...
+				context: 'Theme Showcase filter name',
+			} ),
+			layout: i18n.translate( 'Layout', {
```
Comment (a8ci18n) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **131** times:
`translate( 'Layout' )` *ES Score: 10*


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx [hunk 3]
```diff
@@ -85,6 +85,24 @@ class MagicSearchWelcome extends React.Component {
...
+				context: 'Theme Showcase filter name',
+			} ),
+			column: i18n.translate( 'Columns', {
```
Comment (a8ci18n) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **48** times:
`translate( 'Columns:' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/theme-showcase-filter-translations)


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx [hunk 4]
```diff
@@ -85,6 +85,24 @@ class MagicSearchWelcome extends React.Component {
...
+				context: 'Theme Showcase filter name',
+			} ),
+			subject: i18n.translate( 'Subject', {
```
Comment (a8ci18n) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **136** times:
`translate( 'Subject' )` *ES Score: 12*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/theme-showcase-filter-translations)


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx [hunk 5]
```diff
@@ -85,6 +85,24 @@ class MagicSearchWelcome extends React.Component {
...
+				context: 'Theme Showcase filter name',
+			} ),
+			style: i18n.translate( 'Style', {
```
Comment (a8ci18n) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **135** times:
`translate( 'Style' )` *ES Score: 10*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/theme-showcase-filter-translations)


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx [hunk 6]
```diff
@@ -96,7 +114,10 @@ class MagicSearchWelcome extends React.Component {
 					className="themes-magic-search-card__welcome-taxonomy-icon"
 					size={ 18 }
 				/>
-				{ taxonomy }
+				{ taxonomyTranslations[ taxonomy ] ||
+					i18n.translate( 'Unknown', {
```
Comment (a8ci18n) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **102** times:
`translate( 'Unknown', { context: 'name'} )` *ES Score: 15*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/theme-showcase-filter-translations)


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

**Comment by akirk** [comment]
Current status with D35526-code:

Screenshot 2019-11-29 at 20 39 48

**Comment by akirk** [comment]
translated-showcase

**Comment by akirk** [comment]
This is now ready for review, requires D35526-code.

File: client/components/keyed-suggestions/index.jsx [hunk 7]
```diff
@@ -228,8 +229,18 @@ class KeyedSuggestions extends React.Component {
 				// concatenate mathing and non matchin - this is full set of filters just reordered.
 				filtered[ key ] = [ ...matchingSorted, ...parts[ 1 ] ];
 			} else {
+				// Try a full match first and try substring matches
+				let multiRegex = filterTerm;
```
Comment (deBhal) [inline client/components/keyed-suggestions/index.jsx]: We've got the Fuse library for just this sort of searching, but I had a play and it's more effort that I expected.

It's not expecting the key:value pairs the way we'd expect, 

```node
Fuse = require('fuse.js');
// terms = copy and paste JSON.stringify(terms)
Object.values = Object.values || ((obj) => Object.keys(obj).map((k) => obj[k])) // node doesn't have Object.values
fuse = new Fuse(Object.values(terms.subject), options);
results = fuse.search('Boda')
[ { item: 
     { name: 'Bodas',
       description: 'Con este diseño romántico y organizado, puedes compartir datos importantes, configurar una cuenta atrás para tu gran día y recopilar confirmaciones de asistencia mediante un formulario de contacto.' },
    score: 0.001 },
  { item: 
     { name: 'Anuncio',
       description: 'Temas para bodas, reuniones, lanzamientos de productos: todo lo que quieras anunciar a tus familiares, amigos o clientes. Comparte fotos, fechas, preguntas frecuentes, instrucciones, indicaciones y mucho más.' },
    score: 0.11 },
  { item: 
     { name: 'Moda',
       description: 'Hauls de ropa, entradas del #OOTD, reseñas de desfiles: elige entre una gama de temas elegantes pensados para mostrar fotos y destacar tu feed de Instagram. Algunos son minimalistas y otros tienen toques adicionales de diseño gráfico.' },
    score: 0.1475 },
  { item: 
     { name: 'Colaboración',
       description: 'Temas diseñados para ayudar a los grupos a comunicarse y colaborar con clientes, estudiantes, amigos o compañeros de trabajo, y para planear proyectos, discutir temas urgentes o simplemente charlar.' },
    score: 0.29 },
  { item: 
     { name: 'Comida',
       description: 'Sabrosos temas de WordPress para tu restaurante o blog gastronómico. Publica menús e instrucciones, comparte recetas y fotos de comidas o publica tus reseñas de restaurantes.' },
    score: 0.52 } ]

```

I didn't want to get into fiddling with thresholds, so I think we should go ahead as-is, but if you wanted to circle back later, the docs are at https://fusejs.io/

**Pull request merged**